worlds/yoshisisland/Regions.py

```diff
@@ -0,0 +1,211 @@
+        create_region(multiworld, player, locations_per_region, location_cache, "6-8"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Bowser's Room"),
+
+    ]
+    if get_option_value(multiworld, player, "extras_enabled") == 1:
+        regions.insert(68, create_region(multiworld, player, locations_per_region, location_cache, "6-Extra")),
+        regions.insert(58, create_region(multiworld, player, locations_per_region, location_cache, "5-Extra")),
+        regions.insert(48, create_region(multiworld, player, locations_per_region, location_cache, "4-Extra")),
+        regions.insert(38, create_region(multiworld, player, locations_per_region, location_cache, "3-Extra")),
+        regions.insert(28, create_region(multiworld, player, locations_per_region, location_cache, "2-Extra")),
+        regions.insert(18, create_region(multiworld, player, locations_per_region, location_cache, "1-Extra"))
+
+    if get_option_value(multiworld, player, "minigame_checks") >= 2:
+        regions.insert(74, create_region(multiworld, player, locations_per_region, location_cache, "6-Bonus")),
+        regions.insert(63, create_region(multiworld, player, locations_per_region, location_cache, "5-Bonus")),
+        regions.insert(52, create_region(multiworld, player, locations_per_region, location_cache, "4-Bonus")),
+        regions.insert(41, create_region(multiworld, player, locations_per_region, location_cache, "3-Bonus")),
+        regions.insert(29, create_region(multiworld, player, locations_per_region, location_cache, "2-Bonus")),
+        regions.insert(19, create_region(multiworld, player, locations_per_region, location_cache, "1-Bonus"))
+    multiworld.regions += regions
+
+    connectStartingRegion(multiworld, player)
+
+    bosses = BossReqs(multiworld, player)
+
+    names: Dict[str, int] = {}
+
+    multiworld.get_region('Overworld', player).add_exits(['World 1', 'World 2', 'World 3', 'World 4', 'World 5', 'World 6'],
+                                                        {'World 1': lambda state: state.has('World 1 Gate', player),
+                                                        'World 2': lambda state: state.has('World 2 Gate', player),
+                                                        'World 3': lambda state: state.has('World 3 Gate', player),
+                                                        'World 4': lambda state: state.has('World 4 Gate', player),
+                                                        'World 5': lambda state: state.has('World 5 Gate', player),
+                                                        'World 6': lambda state: state.has('World 6 Gate', player)})
+    CurWorld = 1
+    CurLev = 1
+    for i in range(47):
+        multiworld.get_region(f'World {CurWorld}', player).add_exits([gamevar.level_location_list[i]])
+        if CurLev > 7:
+            CurLev = 1
+            CurWorld += 1
+        else: CurLev += 1

```

Comment:
This might be wrong but there must be another way to do this like this:

```
for i in range(0, 47):
    cur_world = i // 7 + 1
    cur_level = i % 7 + 1
    ...
```

Also, the names CurWorld and CurLev are code style violations. Variable names have to be lower_case_with_underscores. Do you have a PEP 8 extension installed with your IDE?